README
Modify permissions in symfony code to allow symfony 1.4 to run on servers with PHP suexec: dir permissions must not be less restrictive than 0755 and file permissions must not be less restrictive than 0644.

Further information:
--------------------
http://trac.symfony-project.org/wiki/SymfonyPHPsuexec
http://trac.symfony-project.org/ticket/3628

Long discussion here:
http://oldforum.symfony-project.org/index.php/m/45845/index.html

Other symfony bugfixes and patches may be applied. See CHANGELOG for details.

---------------------------------------

Symfony is a complete framework designed to optimize the development of web applications by way of several key features.
For starters, it separates a web application's business rules, server logic, and presentation views.
It contains numerous tools and classes aimed at shortening the development time of a complex web application.
Additionally, it automates common tasks so that the developer can focus entirely on the specifics of an application.
The end result of these advantages means there is no need to reinvent the wheel every time a new web application is built!

Symfony was written entirely in PHP 5.
It has been thoroughly tested in various real-world projects, and is actually in use for high-demand e-business websites.
It is compatible with most of the available databases engines, including MySQL, PostgreSQL, Oracle, and Microsoft SQL Server.
It runs on *nix and Windows platforms.

LICENSE
-------

see LICENSE file

INSTALLATION
------------

see http://www.symfony-project.org/getting-started/1_4/en/


CHANGELOG
CHANGELOG
=========
12/1/13
-------
http://trac.symfony-project.org/ticket/10065
Fix email address with no leading space not linked.
Add option for unicode characters in email username
  /lib/helper/TextHelper.php
  /test/unit/helper/TextHelperTest.php
Thanks to frost-nzcr4

11/19/13
--------
preg_replace /e modifier deprecated as of PHP5.4
  /lib/response/sfWebResponse.class.php
Thanks to https://github.com/jpgerdeman/symfony1/commit/59875eaed9ed060ecc40a788da44179638e38a75

11/18/13
--------
Fix typo in Doctrine implementation of the sfYamlParser
  /lib/plugins/sfDoctrinePlugin/lib/vendor/doctrine/Doctrine/Parser/sfYaml/sfYamlParser.php
Thanks to https://github.com/theodo/symfony1/commit/4de52bff690218a56c03a382b7e339b54d97e321

11/18/13:
---------
Replace all instances of dir & file permissions set to:
  * '0777' => '0755'
  * '0666' => '0644'

List of modified files:
-----------------------
 * /data/bin/sandbox_installer.php
 * /lib/cache/sfFileCache.class.php
 * /lib/cache/sfSQLiteCache.class.php
 * /lib/config/sfConfigCache.class.php
 * /lib/generator/sfGeneratorManager.class.php
 * /lib/i18n/sfMessageSource_gettext.class.php
 * /lib/i18n/sfMessageSource_XLIFF.class.php
 * /lib/log/sfFileLogger.class.php
 * /lib/plugin/sfPearEnvironment.class.php
 * /lib/plugins/sfDoctrinePlugin/lib/generator/sfDoctrineFormFilterGenerator.class.php
 * /lib/plugins/sfDoctrinePlugin/lib/generator/sfDoctrineFormGenerator.class.php
 * /lib/plugins/sfDoctrinePlugin/lib/vendor/doctrine/Doctrine/Migration/Diff.php
 * /lib/plugins/sfDoctrinePlugin/lib/vendor/doctrine/Doctrine/Lib.php
 * /lib/plugins/sfPropelPlugin/lib/generator/sfPropelFormFilterGenerator.class.php
 * /lib/plugins/sfPropelPlugin/lib/generator/sfPropelFormGenerator.class.php
 * /lib/plugins/sfPropelPlugin/lib/task/sfPropelInsertSqlTask.class.php
 * /lib/plugins/sfPropelPlugin/lib/vendor/phing/lib/Zip.php
 * /lib/plugins/sfPropelPlugin/lib/vendor/phing/system/io/FileSystem.php
 * /lib/plugins/sfPropelPlugin/lib/vendor/phing/system/io/PhingFile.php
 * /lib/storage/sfSessionTestStorage.class.php
 * /lib/task/cache/sfCacheClearTask.class.php
 * /lib/task/log/sfLogRotateTask.class.php
 * /lib/task/project/sfProjectPermissionsTask.class.php
 * /lib/task/sfFilesystem.class.php
 * /lib/validator/sfValidatedFile.class.php
 * /lib/vendor/swiftmailer/classes/Swift/FileSpool.php
 * /test/bootstrap/task.php
 * /test/bootstrap/unit.php
 * /test/other/tasksTest.php
 * /test/unit/plugin/sfPearEnvironmentTest.php
 * /test/unit/plugin/sfPearRestPluginTest.php
 * /test/unit/plugin/sfPluginManagerTest.php
 * /test/unit/plugin/sfPluginTestHelper.class.php

11/25/12: Versions 1.4.20
-------------------------

 * [33598] fixed the possibility to fake a file upload
 * [33545] fixed sfPDOSessionStorage for Oracle (closes #10022)
 * [33544] fixed sfWebRequest::splitHttpAcceptHeader incorrect result order (closes #10069, patch by Keri Henare)
 * [33539] fixed exception format when using the PHP 5.4 built-in server (closes #10067, based on a patch from jgskin)
 * [33486] fixed sfPDODatabase::call() method (closes #10044)

10/09/12: Versions 1.4.19
-------------------------

 * [33545] fixed sfPDOSessionStorage for Oracle (closes #10022)
 * [33544] fixed sfWebRequest::splitHttpAcceptHeader incorrect result order (closes #10069, patch by Keri Henare)
 * [33539] fixed exception format when using the PHP 5.4 built-in server (closes #10067, based on a patch from jgskin)
 * [33486] fixed sfPDODatabase::call() method (closes #10044)

05/30/12: Versions 1.4.18
-------------------------

 * [33466] fixed a possible DB session fixation attack (patch from Dmitri Groutso)
 * [33373] fixed test browser click function does not handle css selector without [ or ] (closes #9982, patch from mouette)

03/08/12: Versions 1.4.17
-------------------------

 * [33363] added some tests (closes #3237, patch from Stephen.Ostrow)
 * [33362] fixed a notice on PHP 5.4 (closes #9985, patch from bshaffer)
 * [33358] fixed Notice with PHP 5.4 (closes #10003)
 * [33309] fixed error in Debug mode from sfDebug.class.php when myUser implements sfSecurityUser (closes #9996)
 * [33299] reverted [33226] because of side effects (refs #8348)
 * [33292] fixed test for PHP 5.3 (patch from pylebecq)

12/13/11: Version 1.4.16
------------------------

 * [33251] fixed sfChoiceFormat when a string to translate contains a valid range (closes #9973)
 * [33250] fixed saving i18n fields in subforms (closes #7626, patch from yoshy71)
 * [33249] fixed last modified date calculation in sfMessageSource_Aggregate (closes #9981, patch from jamiel)
 * [33226] fixed merging problem for the routing configuration (closes #8348)
 * [33214] fixed ob_start() behavior on CLI (closes #9970)
 * [33208] fixed ob_start usage (to avoid warning in PHP 5.4, closes #9970)

10/27/11: Version 1.4.15
------------------------

 * [33151] fixed usage of mb_strlen in tasks (closes #9940)
 * [33149] added missing admin.delete_object event (closes #9943)
 * [33137] fixed multiple database support in Propel plugin (for the generator and sfPropelData, closes #8345)
 * [33125] fixed the possibility to include files included in an exclude rule in the deploy task (closes #9912)
 * [33122] fixed include|get_component when sfPartialView class is customized (closes #9932)
 * [33121] fixed protocol relative URL in the asset helper (closes #9936)
 * [33053] fixed typo (closes #9927, based on a patch from pmallet)

09/16/11: Version 1.4.14
------------------------

 * [33025] fixed sfCacheSessionStorage does not always check whether HTTP_USER_AGENT is set (closes #9921, patch from boutell)
 * [33022] fixed auto_link_text() when an email address is already linked (closes #9915, based on a patch from klemens_u)
 * [33021] fixed sfCacheSessionStorage does not serialize properly, also uses wrong option name for httponly (closes #9919, based on a patch from boutell)
 * [32939] fixed memory leak in sfRoute::generate() (closes #9886 - patch from acf0)

lib/log/sfFileLogger.class.php
<?php

/*
 * This file is part of the symfony package.
 * (c) 2004-2006 Fabien Potencier <fabien.potencier@symfony-project.com>
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 */

/**
 * sfFileLogger logs messages in a file.
 *
 * @package    symfony
 * @subpackage log
 * @author     Fabien Potencier <fabien.potencier@symfony-project.com>
 * @version    SVN: $Id: sfFileLogger.class.php 10964 2008-08-19 18:33:50Z fabien $
 */
class sfFileLogger extends sfLogger
{
  protected
    $type       = 'symfony',
    $format     = '%time% %type% [%priority%] %message%%EOL%',
    $timeFormat = '%b %d %H:%M:%S',
    $fp         = null;

  /**
   * Initializes this logger.
   *
   * Available options:
   *
   * - file:        The file path or a php wrapper to log messages
   *                You can use any support php wrapper. To write logs to the Apache error log, use php://stderr
   * - format:      The log line format (default to %time% %type% [%priority%] %message%%EOL%)
   * - time_format: The log time strftime format (default to %b %d %H:%M:%S)
   * - dir_mode:    The mode to use when creating a directory (default to 0755)
   * - file_mode:   The mode to use when creating a file (default to 0644)
   *
   * @param  sfEventDispatcher $dispatcher  A sfEventDispatcher instance
   * @param  array             $options     An array of options.
   *
   * @return Boolean      true, if initialization completes successfully, otherwise false.
   */
  public function initialize(sfEventDispatcher $dispatcher, $options = array())
  {
    if (!isset($options['file']))
    {
      throw new sfConfigurationException('You must provide a "file" parameter for this logger.');
    }

    if (isset($options['format']))
    {
      $this->format = $options['format'];
    }

    if (isset($options['time_format']))
    {
      $this->timeFormat = $options['time_format'];
    }

    if (isset($options['type']))
    {
      $this->type = $options['type'];
    }

    $dir = dirname($options['file']);
    if (!is_dir($dir))
    {
      mkdir($dir, isset($options['dir_mode']) ? $options['dir_mode'] : 0755, true);
    }

    $fileExists = file_exists($options['file']);
    if (!is_writable($dir) || ($fileExists && !is_writable($options['file'])))
    {
      throw new sfFileException(sprintf('Unable to open the log file "%s" for writing.', $options['file']));
    }

    $this->fp = fopen($options['file'], 'a');
    if (!$fileExists)
    {
      chmod($options['file'], isset($options['file_mode']) ? $options['file_mode'] : 0644);
    }

    return parent::initialize($dispatcher, $options);
  }

  /**
   * Logs a message.
   *
   * @param string $message   Message
   * @param string $priority  Message priority
   */
  protected function doLog($message, $priority)
  {
    flock($this->fp, LOCK_EX);
    fwrite($this->fp, strtr($this->format, array(
      '%type%'     => $this->type,
      '%message%'  => $message,
      '%time%'     => strftime($this->timeFormat),
      '%priority%' => $this->getPriority($priority),
      '%EOL%'      => PHP_EOL,
    )));
    flock($this->fp, LOCK_UN);
  }

  /**
   * Returns the priority string to use in log messages.
   *
   * @param  string $priority The priority constant
   *
   * @return string The priority to use in log messages
   */
  protected function getPriority($priority)
  {
    return sfLogger::getPriorityName($priority);
  }

  /**
   * Executes the shutdown method.
   */
  public function shutdown()
  {
    if (is_resource($this->fp))
    {
      fclose($this->fp);
    }
  }
}
